From 355290bfe429c66cd18c2bdd245ae3021d1e2f48 Mon Sep 17 00:00:00 2001
From: Alvaro Lopez <aloga@ifca.unican.es>
Date: Wed, 9 Nov 2011 16:18:39 +0100
Subject: [PATCH] Fixes bug 871877

Added an aditional testunit to the VlanNetworkTestCase to check
if the FixedIpNotFoundForNetwork exception is raised properly.

(cherry picked from commit 6e7b0387e417e4ef36b8ce1e0b7a7db32732838d)

Change-Id: I92a0ca22aadcfa9a7fd434375e239be8ccf4f387
---
 Authors                    |    1 +
 nova/network/manager.py    |    1 +
 nova/tests/test_network.py |   14 ++++++++++++++
 3 files changed, 16 insertions(+), 0 deletions(-)

diff --git a/Authors b/Authors
index 08403b8..ab90964 100644
--- a/Authors
+++ b/Authors
@@ -4,6 +4,7 @@ Adam Johnson <adjohn@gmail.com>
 Ahmad Hassan <ahmad.hassan@hp.com>
 Alex Meade <alex.meade@rackspace.com>
 Alexander Sakhnov <asakhnov@mirantis.com>
+Alvaro Lopez Garcia <aloga@ifca.unican.es>
 Andrey Brindeyev <abrindeyev@griddynamics.com>
 Andy Smith <code@term.ie>
 Andy Southgate <andy.southgate@citrix.com>
diff --git a/nova/network/manager.py b/nova/network/manager.py
index 4fa4d47..b0befab 100644
--- a/nova/network/manager.py
+++ b/nova/network/manager.py
@@ -1025,6 +1025,7 @@ class VlanManager(RPCAllocateFixedIP, FloatingIP, NetworkManager):
             self.db.fixed_ip_associate(context,
                                        address,
                                        instance_id,
+                                       network['id'],
                                        reserved=True)
         else:
             address = kwargs.get('address', None)
diff --git a/nova/tests/test_network.py b/nova/tests/test_network.py
index 32502ea..f7aa684 100644
--- a/nova/tests/test_network.py
+++ b/nova/tests/test_network.py
@@ -309,6 +309,7 @@ class VlanNetworkTestCase(test.TestCase):
         db.fixed_ip_associate(mox.IgnoreArg(),
                               mox.IgnoreArg(),
                               mox.IgnoreArg(),
+                              mox.IgnoreArg(),
                               reserved=True).AndReturn('192.168.0.1')
         db.fixed_ip_update(mox.IgnoreArg(),
                            mox.IgnoreArg(),
@@ -321,6 +322,19 @@ class VlanNetworkTestCase(test.TestCase):
         network['vpn_private_address'] = '192.168.0.2'
         self.network.allocate_fixed_ip(None, 0, network, vpn=True)
 
+    def test_vpn_allocate_fixed_ip_no_network_id(self):
+        network = dict(networks[0])
+        network['vpn_private_address'] = '192.168.0.2'
+        network['id'] = None
+        context_admin = context.RequestContext('testuser', 'testproject',
+                is_admin=True)
+        self.assertRaises(exception.FixedIpNotFoundForNetwork,
+                self.network.allocate_fixed_ip,
+                context_admin,
+                0,
+                network,
+                vpn=True)
+
     def test_allocate_fixed_ip(self):
         self.mox.StubOutWithMock(db, 'fixed_ip_associate_pool')
         self.mox.StubOutWithMock(db, 'fixed_ip_update')
-- 
1.7.6.5

